src/Maestro/tests/Maestro.ScenarioTests/AsyncDisposable.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Maestro.ScenarioTests
+{
+    public class AsyncDisposable : IAsyncDisposable
+    {
+        private readonly Func<ValueTask> _dispose;
+
+        public static IAsyncDisposable Create(Func<ValueTask> dispose)
+        {
+            return new AsyncDisposable(dispose);
+        }
+
+        private AsyncDisposable(Func<ValueTask> dispose)
+        {
+            _dispose = dispose;
+        }
+
+        public ValueTask DisposeAsync()
```

Comment:
Helper things we don't control probably shouldn't be ValueTask, it can be dangerous.

Reply:
This has to be. IAsyncDisposable is `ValueTask DisposeAsync()`

Reply:
Ouch. that's... something. Alrighty then!

Reply:
Yea, its a bit annoying, but it works with `await using (thingy)`